rollout/trafficrouting/istio/istio.go

```diff
@@ -209,17 +209,26 @@ func (r *Reconciler) reconcileVirtualService(obj *unstructured.Unstructured, des
 		}
 	}
 
+	// Generate Patches
 	patches := r.generateVirtualServicePatches(httpRoutes, tlsRoutes, int64(desiredWeight))
 	err = patches.patchVirtualService(httpRoutesI, tlsRoutesI)
 	if err != nil {
 		return nil, false, err
 	}
 
-	err = unstructured.SetNestedSlice(newObj.Object, httpRoutesI, "spec", Http)
-	if err != nil {
-		return newObj, len(patches) > 0, err
+	// Set HTTP Route Slice
+	if len(httpRoutes) > 0 {
+		err = unstructured.SetNestedSlice(newObj.Object, httpRoutesI, "spec", Http)
+		if err != nil {
+			return newObj, len(patches) > 0, err
+		}
 	}
-	err = unstructured.SetNestedSlice(newObj.Object, tlsRoutesI, "spec", Tls)
+
+	// Set TLS Route SLice
+	if len(tlsRoutes) > 0 {
+		err = unstructured.SetNestedSlice(newObj.Object, tlsRoutesI, "spec", Tls)
```

Comment:
missing error handler?

Reply:
I think it'll ultimately be handled  by the return on L#232?